lib/beaker/hypervisor.rb

```diff
@@ -87,6 +88,19 @@ def cleanup
       nil
     end
 
+    #SSH connection method preference. Can be overwritten by hypervisor to change the order
+    def self.connection_preference
+      ['ip', 'vmhostname', 'hostname']
+    end
+
+    def self.set_ssh_connection_perefence(hosts_to_provision, hypervisor)
+      if hypervisor.connection_preference.sort == self.connection_preference.sort
```

Comment:
Using this to test if the array returned by the overriding method in hypervisor class matches the API requirement of having the same element name of `["ip", "vmhostname", "hostname"]`, order does not matter.

I know sorting would bring atleast `n log n` runtime debt, is there any better way to do the same?